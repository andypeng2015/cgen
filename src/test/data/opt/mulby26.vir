module foo

export function w $foo(w %x) {
@start:
  %x = mul.w %x, 26_w
  ret %x
}
