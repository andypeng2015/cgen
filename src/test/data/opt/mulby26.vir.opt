module foo

export function w $foo(w %x) {
@2:
  %5 = lsl.w %x, 0x5_w ; @9
  %6 = lsl.w %x, 0x2_w ; @10
  %8 = lsl.w %x, 0x1_w ; @12
  %7 = sub.w %5, %6 ; @11
  %9 = sub.w %7, %8 ; @13
  ret %9
}
