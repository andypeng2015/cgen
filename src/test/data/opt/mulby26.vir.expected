module foo

export function w $foo(w %x) {
@0:
  %1 = lsl.w %x, 0x5_w ; @3
  %3 = lsl.w %x, 0x2_w ; @5
  %4 = lsl.w %x, 0x1_w ; @6
  %2 = add.w %3, %4 ; @4
  %0 = sub.w %1, %2 ; @2
  ret %0
}
